common/journal_entries/imperia_infamy_rebalance_journal_entries.txt

```diff
@@ -86,31 +77,69 @@ je_is_pariah = {
 		infamy >= infamy_threshold:notorious
 	}
 	immediate = {
+		if = {
+			limit = {
+				NOT = {
+					exists = global_var:current_pariah
+				}
+			}
+			set_global_variable = {
+				name = current_pariah
+				value = this
+			}
+		}
 	}
 	complete = {
 		infamy < infamy_threshold:pariah
 	}
 	on_complete = {
+		remove_modifier ?= modifier_international_pariah
+		remove_modifier ?= modifier_pariah_economic_sanctions
+		remove_modifier ?= modifier_pariah_international_isolation
+		# If theres still another pariah on completion, set the next pariah to be the highest infamy pariah available..
 		if = {
 			limit = {
-				has_modifier = modifier_international_pariah
+				any_country = {
+					has_journal_entry = je_is_pariah
+					this != root
+				}
 			}
-			remove_modifier = modifier_international_pariah
-		}
-		if = {
-			limit = {
-				has_modifier = modifier_pariah_economic_sanctions
+			ordered_country = {
+				limit = {
+					has_journal_entry = je_is_pariah
+					this != root
+				}
+				order_by = infamy
+				position = 0
+				set_global_variable = {
+					name = current_pariah
+					value = this
+				}
```

Comment:
No need, we have fired all the journal entries, the scopes are set. Provided we overwrite the pariah when a new one comes in, we don't need to backtrack.

Reply:
I've been going over the resulting code, and I don't think it would work with all of your changes, which leads me to the conclusion that I might be misunderstanding how you wanted this to work.. Regardless, I've uploaded your wanted changes, except for the monthly_pulse one.